test/unit/contents_test.rb

```diff
@@ -9,13 +9,33 @@ def test_xml_from_contents
     patient = FHIR.from_contents(xml)
     assert !patient.nil?, 'From contents did not succeed with XML.'
     assert patient.id == 'example', 'Patient did not deserialize correctly.'
+    # check memory
+    before = check_memory
+    patient = nil
+    wait_for_gc
+    after = check_memory
+    unless after.empty?
+      puts "BEFORE GC: #{before}"
+      puts "AFTER GC: #{after}"
+    end
+    assert after.empty?, 'Garbage collection missed FHIR Models.'
```

Comment:
Since this section for checking memory is used in most of the tests, we might just put it in a method ... and add it to `teardown`.